### DISCUSSION TITLE
Rebranding

### DISCUSSION THREAD

**Discussion by azinit** [feature-sliced/documentation#399] [Ideas]
Текущее лого FSD создавалось совсем в иных условиях, и совсем для других целей

Внутри кортимы уже давно идут разговоры о том, чтобы актуализировать лого под современные реалии, и даже есть свои наработки

Но если у вас есть свои предложения и идеи - feel free поделиться тут 🤔🍰

**Comment by illright** [comment]
К слову о тортиках. Мне нравится идея ассоциировать feature-sliced с эмоджиком, особенно с тортиком 🍰. Напоминает брендинг [Pinia](https://pinia.vuejs.org/), который теперь ассоциируется с эмоджиком ананаса 🍍, что позволяет [писать о нем красиво](https://github.com/antfu/vitesse#features).

Одной из моих идей для логотипа был нарисованный кусок тортика, сбоку которого виднеются прожилки, как на микросхемах. Вариацией на эту тему может быть три слоя на тортике, каждый со своей символикой. Например, в скетче ниже подразумевается, что средний слой прозрачный, и через него видно внутреннее устройство.

Другой идеей, более банальной, было порубить буквы `fs` и отставить кусочки друг от друга слегка, аки слои. Кусочки бы можно было покрасить, и тогда у нас появился бы колор-кодинг слоев. Нахрена это нужно, спросите вы? Ну, документацию оформить, если цвета в порядке радуги, то еще и сообщать порядок (т.е. понятно, что синий идет после красного). 
[Внимание, не рофл, на серьёзных щах] Ну и еще можно было бы убрать голубой и сделать в цвет ЛГБТ+, добавив на сайтик секцию о том, что коммьюнити FS открыто к самым разным сообществам, а любые оскорбления мы не приветствуем, и за это наказываем. Понимаю, что вопрос очень спорный, так что на усмотрение команды.

<details>
<summary>Скетчи</summary>

![скетчи](https://user-images.githubusercontent.com/15035286/147750299-c80a5881-e91d-4d13-bc39-04bfb60c9935.jpg)

</details>

**Reply by azinit** [reply]
**Спасиб за подробный комментарий и особенно скетчи!**

У кортимы примерно такое же видение и есть похожие скетчи, но твое видение очень сильно дополняет картину для будущей айдентики 🚀

**Reply by ilkiri23** [reply]
Согласен по поводу тортика и эмоджи, но, на мой взгляд, излишняя детализация мешает восприятию  
<details><summary>Парочка идей</summary>

![fsd](https://user-images.githubusercontent.com/96888709/149824290-d1e576a8-345f-4a04-b0a7-8d393b57bb90.png)</p>

**Reply by illright** [reply]
Согласен про детализацию, в моих скетчах явно был перебор.

Подумал насчет вписывания FSD в тортик, чем-то мне это не нравится. Из объективного: из-за этого текст, само название продукта, рядом с лого смотрится странно, как будто одна и та же информация сообщается дважды.

Нарисовал еще вот такое, хочется еще перенести на компьютер и добавить цвета. Здесь тоже есть мелкие детальки, но их мало + они не так важны, поэтому мне кажется, что это лого будет интересно и крупным планом (где-нибудь на лендинге), и понятно мелким планом (в хедере, например).

<details>
<summary>Спойлер</summary>

<img alt="зарисовка тортика с мокапом интерфейса на стороне" width="300" src="https://user-images.githubusercontent.com/15035286/150599908-05a30870-4d9b-4226-a748-f8e04416e434.jpg" />

</details>

**Reply by illright** [reply]
Вот, кстати, моя первая попытка перенести в вектор и раскрасить. Выбор цветов мне не очень нравится, но решил все равно вбросить, чтоб была точка, от которой можно оттолкнуться

<details><summary>Спойлер</summary>

![piece of cake](https://user-images.githubusercontent.com/15035286/152948080-aebd0085-fcdf-4771-9e78-0b94a9c89130.jpg)

</details>

**Comment by azinit** [comment]
## `STATUS-UPDATE:`  03.02.2022

**Обсудили с кортимой основные проблемы текущей айдентики**

Из основного:
- 🍰 Хочется в целом довести айдентику от "уродливой узнаваемости" до "солидного бренда"
- 🎨 Хочется поправить "грязные цвета" в сторону большей солидности, и убрать по-возможности градиент (или сделать его более серьезным)
   (но не навредить узнаваемости)
- 🏷️ Хочется убрать "исторические" ошибки в лого и наконец выбрать что-то из референсов 
   (в том числе тех, что оставили здесь @illright @ilkiri23 )
- 🔠 Начертание склоняется к `FSD`, обязательном слове `Design` в названии и `Feature-Sliced Design` начертании, как официальном от FSD

*Спасибо, что проявляете интерес к теме!*

**Reply by illright** [reply]
Клево! У меня только сомнения насчет дефиса в Feature-Sliced Design. Дефис намекает на то, что Feature-sliced — одно слово, из-за этого капитализировать слово Sliced уже не так красиво. Еще с дефисом могут быть определенные проблемы _неравнозначности_ слов в аббревиатуре, например, вот так уже будет нельзя оформлять текст:
```
Feature
Sliced
Design
```

Так что если этот вопрос обсуждается, я бы отдал голос за начертание `Feature Sliced Design` для названия методологии и `feature-sliced` для тэгов проектов.

**Reply by azinit** [reply]
**Я тоже изначально был за этот вариант без дефиса, но @RinAkaia привела хорошие доводы в сторону дефиса)
(думаю тут отпишет еще подробнее)**

Как минимум, если глянуть сейчас на привычные названия других "подходов-аббревиатур", то, внезапно - там тоже будут дефисы для похожих случаев!

<details>
<summary>За примерами далеко ходить не надо... </summary>

![image](https://user-images.githubusercontent.com/42924400/152949273-a85b2490-43e2-4c28-a01d-3a48b03dd923.png)
![image](https://user-images.githubusercontent.com/42924400/152949289-658b32fa-84a3-46a6-b2db-cae4f19edfc0.png)
![image](https://user-images.githubusercontent.com/42924400/152949298-0d0a7842-37b9-4a3a-aff3-3cf5b9103c75.png)

</details>

<details>
<summary>Хоть и с редкими исключениями</summary>

![image](https://user-images.githubusercontent.com/42924400/152949374-a8a0b4b4-3747-4e10-948a-9dcb72254d1d.png)
![image](https://user-images.githubusercontent.com/42924400/152949391-b0e58d6f-cebb-41a3-9ffe-585927100b9a.png)

</details>

@RinAkaia также со своей лингвистической экспертизой заявляет, что с дефисом и с точки зрения языка [гораздо правильнее](https://www.lexico.com/definition/self-driven) подобное писать

**Reply by illright** [reply]
Проблема с этими подходами-аббевиатурами, как ты абсолютно верно подметил — все пишут их так, как им вздумается. Где можно найти Domain-Driven Design, можно найти и Domain Driven Design.

Насчет лингвистической стороны вопроса — я прогуглил эту тему тут недавно, на English Stack Exchange пишут, что соглашений по таким вопросам в языке нет, и что главное — оставаться консистентными. Не очень понял, на что нужно смотреть в приложенной ссылке на `self-driven`. Однако, я прекрасно понимаю, что Stack Exchange — не истина в последней инстанции, так что если @RinAkaia сможет приложить материалы более уважительных лингвистических инстанций, вопросов никаких не будет)

**Reply by akaia-shadowfox** [reply]
Да, Feature-Sliced было задумало как цельное название и именно с дефисом, чтобы подчеркнуть деепричастный оборот, а буква S в данном случае заглавная потому что Sliced это самостоятельная лексическая единица при переходе в формат аббревиатуры

**Reply by illright** [reply]
> буква S в данном случае заглавная потому что Sliced это самостоятельная морфема

Не очень понимаю, почему второе является причиной первого. Где можно об этом почитать подробнее?

**Reply by akaia-shadowfox** [reply]
Сложно сделать какую-то выкладку по этой теме, здесь проще будет ориентироваться по другим примерам ( https://www.domainlanguage.com/wp-content/uploads/2016/05/DDD_Reference_2015-03.pdf )

Но устоявшееся правило здесь определённо есть, возможно потому что при переходе от акронима обратно к его расшифровке смена first letter case для одного из слов ломает поддержание уровня когнитивной нагрузки

P.S. Изначально ошибочно применила определение "морфема", исправилась

**Reply by akaia-shadowfox** [reply]
Но вообще я считаю, что в нашем случае одна только отсылка к **оригинальной** расшифровке DDD уже самодостаточный аргумент

**Comment by azinit** [comment]
Концепт 🤔 
https://rebrand-sliced.netlify.app/en/
